### PR TITLE
refactor(es/transforms/compat): visitmut for async_to_generator part1

### PR DESCRIPTION
**Pull request by kwonoj**
This PR partially migrates `async_to_generator`'s fold into visit_mut. For easier migration there are some naive choices to copy few things. This may need to be revisited depends on the nature of the copy.

### CODE REVIEW

File: ecmascript/transforms/compat/src/es2017/async_to_generator.rs [hunk 1]
```diff
@@ -970,55 +973,52 @@ struct AsyncFnBodyHandler {
...
                         arg: Some(arg),
                     })
                 } else {
-                    Expr::Yield(YieldExpr {
-                        span,
+                    *expr = Expr::Yield(YieldExpr {
+                        span: *span,
                         delegate: false,
-                        arg: Some(arg),
+                        arg: Some(arg.clone()),
```
Comment (kdy1) [inline ecmascript/transforms/compat/src/es2017/async_to_generator.rs:1010]: Can you change this to `arg.take()`?

File: ecmascript/transforms/compat/src/es2017/async_to_generator.rs [hunk 2]
```diff
@@ -970,55 +973,52 @@ struct AsyncFnBodyHandler {
...
         match expr {
             Expr::Await(AwaitExpr { span, arg }) => {
                 if self.is_async_generator {
                     let callee = helper!(await_async_generator, "awaitAsyncGenerator");
                     let arg = Box::new(Expr::Call(CallExpr {
-                        span,
+                        span: *span,
                         callee,
-                        args: vec![arg.as_arg()],
+                        args: vec![arg.clone().as_arg()],
```
Comment (kdy1) [inline ecmascript/transforms/compat/src/es2017/async_to_generator.rs:998]: Can you change this to `arg.take().as_arg()`?

File: ecmascript/transforms/compat/src/es2017/async_to_generator.rs [hunk 3]
```diff
@@ -1087,15 +1087,15 @@ fn extract_callee_of_bind_this(n: &mut CallExpr) -> Option<&mut Expr> {
...
+fn handle_await_for(stmt: &mut Stmt, is_async_generator: bool) {
     let s = match stmt {
         Stmt::ForOf(
             s @ ForOfStmt {
                 await_token: Some(..),
                 ..
             },
-        ) => s,
-        _ => return stmt,
+        ) => s.clone(),
```
Comment (kdy1) [inline ecmascript/transforms/compat/src/es2017/async_to_generator.rs:1097]: Can you change this to `s.take()`?

**Review by kdy1 — CHANGES_REQUESTED** [review]

File: ecmascript/transforms/compat/src/es2017/async_to_generator.rs (hunk 1 above)
Comment (kwonoj) [reply ecmascript/transforms/compat/src/es2017/async_to_generator.rs]: Sure thing, thanks for the pointer. Updated.

**Review by kdy1 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**